uitest/src/test/java/com/vaadin/tests/themes/valo/ModalWindowTest.java

```diff
@@ -30,7 +34,30 @@ public void modalAnimationsAreDisabled() {
                 is("none"));
     }
 
+    @Test
+    public void modal_curtains_close_correctly() {
+        openTestURL();
+
+        openModalWindow();
+        new Actions(getDriver())
+                .moveToElement(findElement(By.className("v-window-header")))
+                .clickAndHold().sendKeys(Keys.ESCAPE).perform();
```

Comment:
I'd like to see an extra validation step in between starting the drag and pressing escape to see that the curtain is present. Some Drag and Drop cases in the framework require a movement on the mouse before any drag functionality is started.